src/EventListener/LoginListener.php

```diff
@@ -3,33 +3,59 @@
 namespace App\EventListener;
 
 use App\Entity\User;
+use App\Enum\MailCrypt;
 use App\Event\LoginEvent;
 use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\Security\Http\Event\InteractiveLoginEvent;
 use Symfony\Component\Security\Http\SecurityEvents;
+use App\Handler\MailCryptKeyHandler;
 
 readonly class LoginListener implements EventSubscriberInterface
 {
-    public function __construct(private EntityManagerInterface $manager)
-    {
+    private readonly MailCrypt $mailCrypt;
+
+    public function __construct(
+        private EntityManagerInterface $manager,
+        private readonly MailCryptKeyHandler $mailCryptKeyHandler,
+        private readonly int $mailCryptEnv,
+    ) {
+        $this->mailCrypt = MailCrypt::from($this->mailCryptEnv);
     }
 
     public function onSecurityInteractiveLogin(InteractiveLoginEvent $event): void
     {
         $user = $event->getAuthenticationToken()->getUser();
 
         if ($user instanceof User) {
-            $this->handleLogin($user);
+            $password = $event->getRequest()->get('_password');
```

Comment:
Is this guaranteed to work on all interactive logins? I don't know the details of this `InteractiveLoginEvent` and when exactly it is dispatched. Is login via the login form the only interactive login scenario? I guess there `_password` is accessible in the request.